.github/copilot-instructions.md

```diff
@@ -0,0 +1,145 @@
+# Ontology of Biological Attributes (OBA) Project Guide
+
+This includes instructions for editing the OBA ontology. 
+
+## Project Layout
+- Main development file is `src/ontology/oba-edit.obo`
+- ODK and ontology documentation can be found in `/docs/`
+
+## Querying ontology
+
+- To look at a specific term if you know the ID:
+    - `obo-grep.pl -r 'id: OBA:2040177' src/ontology/oba-edit.obo`
+- All mentions of an ID
+    - `obo-grep.pl -r 'OBA:2040177' src/ontology/oba-edit.obo`
+- Note that `obo-grep.pl` is in your PATH, no need to search for it    
+- Only search over `src/ontology/oba-edit.obo`
+- DO NOT bother doing your own greps over the file, or looking for other files, unless otherwise asked, you will just waste time.
+- ONLY use the methods above for searching the ontology
+
+## Before making edits
+- Read the request carefully and make a plan, especially if there is nuance
+- If related issues are mentioned read them: `gh issue view GITHUB-ISSUE-NUMBER`
+- If a PMID is mentioned in the issue, ALWAYS try and read it
+- ALWAYS check proposed parent terms for consistency
+
+## Editors guide
+
+### How to create new terms in OBA
+
+#### Overview
+- IMPORTANT: Do not edit the edit file directly, it's large
+- OBA terms are compositional, check DOSDP design patterns: `src/patterns/dosdp-patterns/*.yaml`
+- Desgin pattern templates are found in `src/patterns/data/default`
+
+#### Steps
+
+    1- Read the issue carefully to identify design pattern filler terms.
+
+        Example:
+
+            ```
+            ```
+
+    2- Check whether design pattern components have been imported to OBA. Search `src/ontology/imports/merged_import.owl`
+
+    3- If they do, please proceed to creating the new OBA term using the appropriate DOSDP pattern template.
+
+    4- Add the term to the template, then run 
+            sh run.sh make IMP=false MIR=false ../patterns/definitions.owl 
+
+    5- Refresh OBA imports by running
+            sh run.sh make refresh-merged 
+
+    6 - Run sh run.sh make IMP=false MIR=false ../patterns/definitions.owl again.
+
+    7- When creating new OBA terms using DOSDP patterns, it may be necessary to import terms from other ontologies like ChEBI, the Chemical Entities of Biological Interest or PRO, the PRotein Ontology. However, they are too large to be managed as standard imports so are managed as slims. When a new term you are using in a DOSDP pattern is not yet in a slim you will have to refresh the slim first. They are located here:
```

Comment:
If a term does not exist in pro_obo_slim, ncbitaxon or chebi_obo_slim, make a comment in the issue asking these terms to be added manually.